File: app/src/main/java/net/ginteam/carmen/network/api/CategoryService.java
package net.ginteam.carmen.network.api;

import net.ginteam.carmen.model.ResponseModel;
import net.ginteam.carmen.model.category.CategoryModel;

import java.util.List;

import retrofit2.http.GET;
import rx.Observable;

/**
 * Created by Eugene on 12/23/16.
 */

public interface CategoryService {

    @GET(ApiLinks.CATALOG.CATEGORIES)
    Observable <ResponseModel<List<CategoryModel>>> fetchCategories();

}

File: app/src/main/kotlin/net/ginteam/carmen/kotlin/api/service/CategoryService.kt
package net.ginteam.carmen.kotlin.api.service

import net.ginteam.carmen.kotlin.api.ApiSettings
import net.ginteam.carmen.kotlin.manager.ApiManager
import net.ginteam.carmen.kotlin.model.CategoryModel
import net.ginteam.carmen.kotlin.model.ResponseModel
import retrofit2.http.GET
import rx.Observable

/**
 * Created by eugene_shcherbinock on 2/9/17.
 */
interface CategoryService {

    @GET(ApiSettings.Catalog.GET_CATEGORIES)
    fun fetchCategories(): Observable <ResponseModel <List <CategoryModel>>>

    companion object {
        fun create(): CategoryService = ApiManager.retrofit.create(CategoryService::class.java)
    }

}